src/shaders/line_pattern.vertex.glsl

```diff
@@ -96,4 +100,8 @@ void main() {
     v_linesofar = a_linesofar;
     v_width2 = vec2(outset, inset);
     v_width = floorwidth;
+
+#if defined( FOG ) && !defined( RENDER_TO_TEXTURE )
+    v_fog_pos = fog_position(pos);
+#endif
```

Comment:
Probably overengineering, but we could introduce something like 
```
#pragma mapbox: inline fog_position<pos>
```
to replace these copy-pasta'd chunks chunks

It'd be very similar to our existing pragma expansion in `shaders.js#compile(..)` except it'd inline named chunks of code with some templating. The `fog_pos` chunk can be implemented as
```
export function fogPosChunk(positionVar: string): string {
   return `#if defined( FOG ) && !defined( RENDER_TO_TEXTURE )
         v_fog_pos = fog_position(${positionVar});
    #endif`;
}
```

Reply:
It might be moving out the context a bit much. If we have v_fog_pos is in prelude, this in shaders.js#compile(...), we're saving a few lines but it becomes non-trivial to trace back where it's coming from.

How about this rule: have uniforms in prelude _only_ if their immediate use is in a global functions defined in the prelude and leave varyings out of any preludes as they require an explicit assignment from shader code?

Reply:
@arindam1993 What do you think of, for now, undefining `FOG` when `RENDER_TO_TEXTURE` is defined, so that the preprocessor statement shortens to `#ifdef FOG` and is much less burdensome/repetitive. And if we want to design a separate mechanism, we can address that in followup.

Reply:
For now, I've followed this comment and opted to undefine FOG when rendering to texture: https://github.com/mapbox/mapbox-gl-js/pull/10496#discussion_r599705839